### PR TITLE
Move cxgtool into CLI and modularize conversion functions

### PR DESCRIPTION
**Pull request by maniarathi**

### CODE REVIEW

File: server/cli/convert_to_cxg.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+@click.option(
+    "-s",
+    "--sparse-threshold",
+    help="If the dataset's percent of non-zero values falls belows the specified threshold, then the X "
+         "array of the dataset will be sparse. Since the default value is 0.0, the default will be to "
+         "convert to dense array.",
+    default=0.0,
+    show_default=True,
+)
+@click.option("--obs-names", help="The name of the observation annotations.")  # TODO expand this
```
Comment (MDunitz) [inline server/cli/convert_to_cxg.py:58]: todo

File: server/cli/convert_to_cxg.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+@click.option("--var-names", help="The name of the variables annotations.")  # TODO expand this
```
Comment (MDunitz) [inline server/cli/convert_to_cxg.py:59]: todo

File: server/cli/convert_to_cxg.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+@click.option(
+    "--disable-custom-colors",
+    help="When set, conversion process will not extract scanpy-compatible category colors from the H5AD " "file.",
```
Comment (mckinsel) [inline server/cli/convert_to_cxg.py:62]: Little extra space there

File: server/cli/convert_to_cxg.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    h5ad_data_file.to_cxg(cxg_output_container, sparse_threshold,
+                          convert_anndata_colors_to_cxg_colors=not disable_custom_colors)
+
+
+def get_output_directory(input_filename, output_directory, should_overwrite):
+    """
+    Get the name of the CXG output directory to be created/populated during the dataset conversion.
+    """
+
+    if not path.isfile(output_directory) or (path.isfile(output_directory) and should_overwrite):
```
Comment (mckinsel) [inline server/cli/convert_to_cxg.py:116]: Should these be `isdir`?

File: server/cli/convert_to_cxg.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        var_names,
+        disable_custom_colors,
+        disable_corpora_schema,
+        should_overwrite,
+):
+    """
+    Convert a dataset file into CXG.
+    """
+
+    h5ad_data_file = H5ADDataFile(input_file, backed, title, about, obs_names, var_names, not disable_corpora_schema)
```
Comment (MDunitz) [inline server/cli/convert_to_cxg.py:102]: `not disable_corpora_schema` seems a little odd, maybe you could add in the arg names to make it clear why you are reversing the boolean?

File: server/common/utils/cxg_generation_utils.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+    This function is primarily used to save metadata about a dataset to the CXG directory. At some point, tiledb will
+    have support for metadata on groups at which point the utility of this function should be revisited. Until such
+    feature exists, this function create an empty array and annotate that array.
+
+    For more information, visit https://github.com/TileDB-Inc/TileDB-Py/issues/254.
+    """
+
+    array_name = f"{cxg_container}/{group_metadata_name}"
+    with tiledb.from_numpy(array_name, np.zeros((1,))):
```
Comment (mckinsel) [inline server/common/utils/cxg_generation_utils.py:21]: What's this do?

Comment (maniarathi) [reply server/common/utils/cxg_generation_utils.py]: I added a comment to explain but basically you need to create an empty array first and then attach the metadata onto the array. This line just creates the empty array.

File: server/cli/convert_to_cxg.py (hunk 4 above)
Comment (maniarathi) [reply server/cli/convert_to_cxg.py]: Good catch! Fixed!

File: server/cli/convert_to_cxg.py (hunk 5 above)
Comment (maniarathi) [reply server/cli/convert_to_cxg.py]: Great suggestion! Done.

File: server/cli/convert_to_cxg.py (hunk 3 above)
Comment (maniarathi) [reply server/cli/convert_to_cxg.py]: Done

File: server/cli/convert_to_cxg.py (hunk 1 above)
Comment (maniarathi) [reply server/cli/convert_to_cxg.py]: Done

File: server/cli/convert_to_cxg.py (hunk 2 above)
Comment (maniarathi) [reply server/cli/convert_to_cxg.py]: Thanks for pointing out. Done.

**Pull request merged**